### PR TITLE
refactor: Simplify fetch-actor-details — deduplicate handler, colocate schemas, clean utils

### PR DESCRIPTION
**Pull request by jirispilka**
Collapses the refactor chain #690 → #691 → #692 → #693 → #694 into a single PR. Closes #690, #691, #692, #693.

## What changed

Five focused refactoring steps across `fetch-actor-details` and its supporting utils:

1. **Deduplicate handler logic** (#690) — extract shared `buildFetchActorDetailsResult` so default and openai variants share one handler instead of duplicating it.
2. **Simplify `buildActorDetailsTextResponse` signature** (#691) — remove pass-through args; caller pre-resolves `mcpToolsMessage` before calling the (now pure/sync) builder.
3. **Colocate schemas and response builders** (#692) — move `actorDetailsOutputOptionsSchema`, `resolveOutputOptions`, `buildActorDetailsTextResponse`, and `buildActorNotFoundResponse` into `fetch_actor_details_common.ts` next to the handler that owns them.
4. **Apply dedent / `texts[]` array convention** (#693) — replace template-literal string concatenation with `dedent` + array `.join('\n')` consistently.
5. **Simplify `actor_details` utils** (#694) — collapse `typeObjectToString`/`typeValueToString` (35 → 13 lines), cache `apifyClient.actor()`, rename `processActorDetailsForResponse` → `buildActorDetailsForWidget`, remove dead return fields.

No behavior change for callers. The public API shape (`output` parameter schema, all response fields) is identical to master.

## Test plan
- [x] `npm run type-check` passes
- [x] `npm run lint` passes
- [x] `npm run test:unit` — 498 tests pass (17 new tests for `typeObjectToString` edge cases)
- [x] Integration tests (requires APIFY_TOKEN, human-run)
- [x] E2E regression results
- [x] Tested UI widgets

## E2E regression results

Tested against production using native MCP client calls — `apify-dev` (this branch) vs `apify-prod` (master) — running all scenarios in parallel.

| Scenario | Result |
|----------|--------|
| Default call (all fields) | ✅ identical |
| `inputSchema: true` | ✅ identical |
| `readme: true` | ✅ identical |
| `description: true` | ✅ identical |
| `stats: true` | ✅ identical |
| `pricing: true` | ✅ identical |
| `rating: true` | ✅ identical |
| `metadata: true` | ✅ identical |
| `mcpTools: true` (MCP Actor) | ✅ identical |
| Non-existent actor → error message | ✅ identical |
| README summary fallback (cheerio-scraper) | ✅ identical |
| `outputSchema: true` | ⚠️ infrastructure gap only — dev server has no `actorStore`, prod does; code path identical |

**Zero behavioral regressions.**

## Widget testing

image

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tests/unit/utils.actor_details.test.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            tags: ['string'],
+            meta: { id: 'string' },
+            flag: false,
+        })).toBe('{ name: string, tags: string[], meta: { id: string } }');
+    });
+
+    it('emits "unknown" for function / symbol / bigint nested inside an array', () => {
+        expect(typeObjectToString({ fns: [() => 1] as unknown[] })).toBe('{ fns: unknown[] }');
+        expect(typeObjectToString({ syms: [Symbol('x')] as unknown[] })).toBe('{ syms: unknown[] }');
+    });
```
[the remark this answers is not in the document]
Comment (jirispilka) [reply tests/unit/utils.actor_details.test.ts]: @copilot the big int was a mistake, let's remote it from tests
